### PR TITLE
Implement book editing

### PR DESCRIPTION
**Pull request by Camotoy**
Updates the PR created by @ForceUpdate1 for 1.16 support. Seems to work fine now that hand support is in MCProtocolLib.

### CODE REVIEW

**Comment by DoctorMacc** [comment]
!start-test-server

**Comment by toinouH** [comment]
It doesn't fixe the problem for the moment

**Comment by DoctorMacc** [comment]
!start-test-server

**Comment by DoctorMacc** [comment]
!stop-test-server

**Comment by DoctorMacc** [comment]
!start-test-server

**Comment by DoctorMacc** [comment]
!start-test-server

**Comment by DoctorMacc** [comment]
!stop-test-server

**Comment by DoctorMacc** [comment]
!start-test-server

**Comment by DoctorMacc** [comment]
!stop-test-server

**Comment by DoctorMacc** [comment]
!start-test-server

**Comment by DoctorMacc** [comment]
!stop-test-server

**Comment by DoctorMacc** [comment]
!start-test-server

**Comment by DoctorMacc** [comment]
This PR is going to be on hiatus for a while.

The problem is that Bedrock sends a book edit update multiple times, while Java just wants it after the book is finished editing. This is fine on vanilla, but server software such as Paper checks for quick book editing and kicks the user. Unlike signs, there is no final packet sent. And we can't just send the last book edit packet because there are different kinds of packets being sent.

**Comment by DoctorMacc** [comment]
Thanks to @davchoo, this PR is no longer on hiatus.

**Comment by DoctorMacc** [comment]
!start-test-server

**Comment by Hellohi3654** [comment]
How well does this work?

**Comment by Hellohi3654** [comment]
any major bugs?

**Comment by toinouH** [comment]
@Hellohi3654 We had some issues in the past with it so.

**Comment by Camotoy** [comment]
I intend on merging into master soon.

**Comment by Hellohi3654** [comment]
so should be fine to merge into my fork

File: connector/src/main/java/org/geysermc/connector/network/session/cache/BookEditCache.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2019-2020 GeyserMC. http://geysermc.org
```
Comment (rtm516) [inline connector/src/main/java/org/geysermc/connector/network/session/cache/BookEditCache.java:2]: Needs updating to 2021
```suggestion
 * Copyright (c) 2019-2021 GeyserMC. http://geysermc.org
```

File: connector/src/main/java/org/geysermc/connector/network/translators/bedrock/BedrockBookEditTranslator.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2019-2020 GeyserMC. http://geysermc.org
```
Comment (rtm516) [inline connector/src/main/java/org/geysermc/connector/network/translators/bedrock/BedrockBookEditTranslator.java:2]: Same as above
```suggestion
 * Copyright (c) 2019-2021 GeyserMC. http://geysermc.org
```

**Review by rtm516 — CHANGES_REQUESTED** [review]

**Review by rtm516 — APPROVED** [review]

**Pull request merged**